src/win/tty.c

```diff
@@ -57,11 +57,17 @@
 
 static void uv_tty_capture_initial_style(CONSOLE_SCREEN_BUFFER_INFO* info);
 static void uv_tty_update_virtual_window(CONSOLE_SCREEN_BUFFER_INFO* info);
+static int uv_cancel_read_console(uv_tty_t* handle);
 
 
 /* Null uv_buf_t */
 static const uv_buf_t uv_null_buf_ = { 0, NULL };
 
+static CRITICAL_SECTION uv_read_console_trap_lock;
+static volatile BOOL uv_read_console_trap = FALSE;
+static volatile BOOL uv_read_console_started = FALSE;
+static HANDLE uv_read_console_done = NULL;
```

Comment:
Maybe rename this to ... done_event?  Just "done" sounded boolean to me. Also, maybe initialize it to INVALID_HANDLE_VALUE?